### PR TITLE
Add CMake configuration

### LINKED ISSUE #2: Add CMake configuration
I find CMake much easier to deal with and maintain than autotools. It also supports tools like ninja.

CMake would also allow us to easily build on Windows and get rid of the custom build setup that exists over there (assuming we want to keep maintaining Windows support).

### PR DESCRIPTION
**Pull request by whittlec**
Simple CMake configuration to make building on Windows easier.

resolves #2

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+check_include_file("sys/stat.h" HAVE_SYS_STAT_H)
+check_include_file("sys/time.h" HAVE_SYS_TIME_H)
+check_include_file("sys/types.h" HAVE_SYS_TYPES_H)
+check_include_file("sys/wait.h" HAVE_SYS_WAIT_H)
+check_include_file("unistd.h" HAVE_UNISTD_H)
+
+configure_file(config-cmake.h.in config.h @ONLY)
+
+if (${CMAKE_HOST_SYSTEM_NAME} MATCHES "Windows")
+    add_definitions( -D__WIN32__)
```
Comment (kadler) [inline CMakeLists.txt:29]: We should probably change the source to use the standard `_WIN32` instead.

**Review by kadler — COMMENTED** [review]
Thanks! Looks pretty good. If you could fix the one suggestion and add ending newlines to the new files, that'd be great.

File: CMakeLists.txt (hunk 1 above)
Comment (whittlec) [reply CMakeLists.txt]: I've also defined `_WIN32` but left the existing `__WIN32__` as there is code that already depends on that - it won't build if only `_Win32` is defined. I don't want to destabilize the existing build system.

Comment (kadler) [reply CMakeLists.txt]: Sorry, I should have been more clear. What I meant was to modify the source files that use `__WIN32__` to use `_WIN32` instead so it builds without this extra definition. Windows compilers already predefine `_WIN32` so there's no point in defining it ourselves. Also, the existing Windows system is dead and I was going to remove it anyway.

I can make the `_WIN32` changes though and clean that up after if you just want me to merge as-is, though.

**Pull request merged**